src/types.ts

```diff
@@ -1,6 +1,7 @@
+import type { MODE_CHOICES } from './constants'
 import type { SortOption } from './utils/sort'
 
-export type RangeMode = 'default' | 'major' | 'minor' | 'patch' | 'latest' | 'newest'
+export type RangeMode = typeof MODE_CHOICES[number]
```

Comment:
I prefer not to specify the literal type twice, so I employ a minor trick to infer the type from the constant variable.